source/Ship.cpp

```diff
@@ -1318,7 +1319,7 @@ void Ship::SetHail(const Phrase &phrase)
 
 string Ship::GetHail(const PlayerInfo &player) const
 {
-	map<string, string> subs;
+	map<string, string> subs = GameData::GetTextReplacements().Substitutions(player);
```

Comment:
I can imagine that this function becomes expensive when we have a lot of conditional replacements. Would it make sense to retrieve the hailStr from line 1333/1334 first (and only calculate the subs if the hailStr is not empty)?